src/main/java/io/strimzi/kafka/access/KafkaAccessReconciler.java

```diff
@@ -4,35 +4,130 @@
 import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
 import io.javaoperatorsdk.operator.api.reconciler.EventSourceInitializer;
 import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
 import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
 import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
+import io.strimzi.api.kafka.model.Kafka;
+import io.strimzi.kafka.access.model.BindingStatus;
 import io.strimzi.kafka.access.model.KafkaAccess;
+import io.strimzi.kafka.access.model.KafkaAccessStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import java.util.Collections;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 
-@ControllerConfiguration
+import static io.javaoperatorsdk.operator.api.reconciler.Constants.NO_FINALIZER;
+
+@ControllerConfiguration(finalizerName = NO_FINALIZER)
 public class KafkaAccessReconciler implements Reconciler<KafkaAccess>, EventSourceInitializer<KafkaAccess> {
 
+    private static final String TYPE_SECRET_KEY = "type";
+    private static final String TYPE_SECRET_VALUE = "kafka";
+    private static final String PROVIDER_SECRET_KEY = "provider";
+    private static final String PROVIDER_SECRET_VALUE = "strimzi";
+    private static final String SECRET_TYPE = "servicebinding.io/kafka";
+
     private final KubernetesClient kubernetesClient;
+    private final Map<String, String> commonSecretData = new HashMap<>();
+    private final Map<String, String> commonSecretLabels = new HashMap<>();
 
-    public KafkaAccessReconciler(KubernetesClient kubernetesClient) {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaAccessOperator.class);
+
+    public KafkaAccessReconciler(final KubernetesClient kubernetesClient) {
         this.kubernetesClient = kubernetesClient;
+        final Base64.Encoder encoder = Base64.getEncoder();
+        commonSecretData.put(TYPE_SECRET_KEY, encoder.encodeToString(TYPE_SECRET_VALUE.getBytes(StandardCharsets.UTF_8)));
+        commonSecretData.put(PROVIDER_SECRET_KEY, encoder.encodeToString(PROVIDER_SECRET_VALUE.getBytes(StandardCharsets.UTF_8)));
+        commonSecretLabels.put(Utils.MANAGED_BY_LABEL_KEY, Utils.KAFKA_ACCESS_LABEL_VALUE);
     }
 
     @Override
-    public UpdateControl<KafkaAccess> reconcile(KafkaAccess kafkaAccess, Context context) {
-        return UpdateControl.noUpdate();
+    public UpdateControl<KafkaAccess> reconcile(final KafkaAccess kafkaAccess, final Context context) {
+        final String kafkaAccessName = kafkaAccess.getMetadata().getName();
+        final String kafkaAccessNamespace = kafkaAccess.getMetadata().getNamespace();
+        LOGGER.info(String.format("Reconciling KafkaAccess %s/%s", kafkaAccessNamespace, kafkaAccessName));
```

Comment:
why not using the `{}` placeholder instead of `String.format`? So ...

```java
LOGGER.info("Reconciling KafkaAccess {}/{}", kafkaAccessNamespace, kafkaAccessName);
```

Reply:
Good point, I'd forgotten you could do that